calico_containers/adapter/datastore.py

```diff
@@ -716,6 +792,51 @@ def get_ep_id_from_cont(self, hostname, container_id):
                 "Container with ID %s has no endpoints." % container_id)
 
     @handle_errors
+    def query_endpoint(self, endpoint_id=None, workload_id=None, orchestrator_id=None, hostname=None):
```

Comment:
the original function returns a dictionary of values where the keys provide the hostname and workdload id - perhaps something to do with that.